openedx/core/djangoapps/coursegraph/management/commands/dump_to_neo4j.py

```diff
@@ -168,18 +164,55 @@ def add_to_transaction(neo4j_entities, transaction):
             transaction.create(entity)
 
     @staticmethod
-    def should_dump_course(course_key):
+    def get_command_last_run(course_key, graph):
+        """
+        This information is stored on the course node of a course in neo4j
+        :param course_key: a CourseKey
+        :param graph: a py2neo Graph
+        :return: a string representation of when this course was last dumped to
+         neo4j or None
+        """
+        selector = NodeSelector(graph)
+        course_node = selector.select(
+            "course",
+            course_key=six.text_type(course_key)
+        ).first()
+
+        last_this_command_was_run = None
+        if course_node is not None:
+            last_this_command_was_run = course_node['time_last_dumped_to_neo4j']
+
+        return last_this_command_was_run
+
+    @staticmethod
+    def get_course_last_published(course_key):
+        """
+        We use the CourseStructure table to get when this course was last
+          published.
+        :param course_key: a CourseKey
+        :return: a string representation of when this course was last dumped to
+          neo4j or None
+        """
+        try:
+            structure = CourseStructure.objects.get(course_id=course_key)
+            last_course_had_published_event = six.text_type(structure.modified)
+        except CourseStructure.DoesNotExist:
+            last_course_had_published_event = None
+
+        return last_course_had_published_event
+
+    def should_dump_course(self, course_key, graph):
```

Comment:
It's not a very long method; if I rewrite it to only have one return, I imagine it would look like:

```python
       last_this_command_was_run = self.get_command_last_run(course_key, graph)

        course_last_published_date = self.get_course_last_published(course_key)

        should_dump = False
        # if we don't have a record of the last time this command was run,
        # we should serialize the course
        if last_this_command_was_run is None:
            should_dump = True

        # if we've serialized the course recently and we have no published
        # events, we can skip re-serializing it
        elif last_this_command_was_run and course_last_published_date is None:
            should_dump = False

        # otherwise, serialize if the command was run before the course's last
        # published event
        else:
            should_dump = last_this_command_was_run < course_last_published_date

        return should_dump
```

I don't see how that's any cleaner; it adds more lines and feels more cluttered